### PR TITLE
Fix: Changed profile.username to profile.displayName for header username display

### PR DESCRIPTION
**Pull request by Junjiequan**
## Description

This PR adapts to the https://github.com/SciCatProject/scicat-backend-next/pull/1041
In the above backend PR we have changed `profile.username` to store `oidc.sub` by default, which in some cases might return string of numbers instead of user's name. So the change of displaying `profile.username` to` profile.displayName` on the header section is required.

## Tests included/Docs Updated?

- [ ] Included for each change/fix?
- [ ] Passing? (Merge will not be approved unless this is checked) 
- [ ] Docs updated?
- [ ] New packages used/requires npm install? 
- [ ] Toggle added for new features?
- [ ] Requires update of SciCat backend API?

### CODE REVIEW

File: src/app/state-management/selectors/user.selectors.ts [hunk 1]
```diff
@@ -22,7 +22,7 @@ export const selectCurrentUserName = createSelector(
   selectCurrentUser,
   (profile, user) => {
     if (profile) {
-      return profile.username.replace("ms-ad.", "");
+      return profile.displayName.replace("ms-ad.", "");
```
Comment (bpedersen2) [inline src/app/state-management/selectors/user.selectors.ts:25]: the replace is probably not needed here then.  The backend prependes a marker for the auth source to the username, but  should not do that for the displayName.

**Review by bpedersen2 — CHANGES_REQUESTED** [review]
Good idea, could get some simplification. 

Actually the `replace(' ms-ad',...`) is probably wrong at all places,  as e.g. the `oidc.` prefix would need the same handling. 

Users without a profile should be limited to the system accounts, if I am not mistaken.

File: src/app/state-management/selectors/user.selectors.ts (hunk 1 above)
Comment (bpedersen2) [reply src/app/state-management/selectors/user.selectors.ts:25]: I  checked and all user types do implement the user profile (local, oidc, ldap) . jwt auth relies on the other strategies for initial login/account creation, so we proably can rely on the profile being there (or adopt the BE to ensure that),

Comment (nitrosx) [reply src/app/state-management/selectors/user.selectors.ts:25]: @bpedersen2 hopefully I understood your comment above correctly. 
To clarify further: user profile is created for all accounts, no matter which auth strategy they use. It is safe to assume that profile is always present.
I agree with you that the displayName should not be stored in the db with a prefix.
We should display it as it is saved in the database.

**Review by nitrosx — CHANGES_REQUESTED** [review]

**Review by nitrosx — APPROVED** [review]

**Review by bpedersen2 — APPROVED** [review]

**Pull request merged**